### DISCUSSION TITLE
Is import functionality necessary?

### DISCUSSION THREAD

**Discussion by savchenko** [taskchampion/taskchampion#162] [Q&A]
Inspired by https://github.com/djmitche/taskchampion/issues/94.

If user's task list is simple and short enough, they probably can close tasks in their current TODO list and create new ones in TaskChampion. 

If it is long and convoluted, then parsing it correctly might become a non-trivial task with a high chance of attracting negative sentiment from those experiencing obscure bugs.

Think about "It has missed sub-task number 183!", "categories and projects get mixed-up if tasks have the same tag!" and so on.

**Comment by djmitche** [comment]
Good question -- is doing a partial job better or worse than nothing?

**Reply by savchenko** [reply]
I'd vote for "less, but better" in this particular case. @dbr , do you have an opinion on this one?

**Reply by dbr** [reply]
I agree that converting non-trivial task lists using lots of TW functionality is probably too hard to do properly (if the concepts don't line up 1:1 then the import process becomes somewhat subjective)

Having the import logic in a separate project (maybe maintained by someone else) seems better as it doesn't tie our release mechanism to the other project (e.g if they make a breaking change in their export API)

I think a good compromise might be to have a `ta import` command symmetrical to https://github.com/taskchampion/taskchampion/issues/173 - then someone could write a tw->tc import script which does something like `tw export | taskwarrior-to-taskchampion | ta import`

**Reply by djmitche** [reply]
I commented #173 to that effect.
